src/editor/app.js

```diff
@@ -390,6 +390,26 @@ UiDriver.registerEventHandler("C_CMD_SET_THEME", function(msg, data, prevReturn)
     editor.setOption("theme", data.name);
 });
 
+UiDriver.registerEventHandler("C_CMD_SET_FONT", function(msg, data, prevReturn) {
+
+    var fontSize = (data.size != "" && data.size>0) ? ("font-size:" + data.size + "px;") : "";
+
+    var fontFamily = data.family != "" ? ("font-family:'" + data.family + "';") : "";
+
+    var styleTag = document.getElementById('userFont');
+
+    if( styleTag  ){ 
+        styleTag.innerHTML = ".editor > .CodeMirror { " + fontFamily + fontSize + " }";
+        styleTag.appendTo('head');
```

Comment:
> appendTo is not a member function of styleTag, it actually is a jQuery function, so right now the only thing it does is to throw an error :)

Welp, goes to show that my interactions with javascript generally involve a lot of duct tape and luck.

> Can you explain it again?

So this is how I understood it: the C_CMD_SET_THEME call appends its link tag to <head>, then C_CCMD_SET_FONT appends its style tag. So after that the end of <head> looks like this:

        ....
        <link type="css" href="theme.css" blah blah/>
        <style> .editor > .CodeMirror { font-family: "this"; font-size: "that" } </style>
    </head>

If the user changes the theme afterwards, C_CMD_SET_THEME is called again and the updates link tag is again appended to head:

        ....
        <style> .editor > .CodeMirror { font-family: "this"; font-size: "that" } </style>
        <link type="css" href="new_theme.css" blah blah/>
    </head>

However, if new_theme.css modifies font-family or font-size on their own (only a few themes do that), the data from the <style> tag will be overridden since new_theme.css comes after it. That's why I'm calling C_CMD_SET_FONT again after any change of themes, with the intention to move the <style> tab down again.